services/console-proxy/server/src/com/cloud/consoleproxy/vnc/VncClient.java

```diff
@@ -82,22 +91,36 @@ public static void main(String args[]) {
     }
 
     private static void printHelpMessage() {
-        /* LOG */s_logger.info("Usage: HOST PORT PASSWORD.");
+        /* LOG */
+        s_logger.info("Usage: HOST PORT PASSWORD.");
     }
 
-    public VncClient(ConsoleProxyClientListener clientListener) {
-        noUI = true;
-        this.clientListener = clientListener;
-    }
-
-    public VncClient(String host, int port, String password, boolean noUI, ConsoleProxyClientListener clientListener) throws UnknownHostException, IOException {
-
-        this.noUI = noUI;
-        this.clientListener = clientListener;
-        connectTo(host, port, password);
+    /**
+     * Reverse bits in byte, so least significant bit will be most significant
+     * bit. E.g. 01001100 will become 00110010.
+     * <p>
+     * See also: http://www.vidarholen.net/contents/junk/vnc.html ,
+     * http://bytecrafter
+     * .blogspot.com/2010/09/des-encryption-as-used-in-vnc.html
+     *
+     * @param b a byte
+     * @return byte in reverse order
+     */
+    private static byte flipByte(byte b) {
```

Comment:
I have picked this piece of code from ajax Implementation. Will check if it already has a unit test. If not i'll write a new one